model/TaskLogBroker/RdsTaskLogBroker.php

```diff
@@ -225,6 +228,109 @@ public function getReport($taskId)
     }
 
     /**
+     * @inheritdoc
+     */
+    public function findAvailableByUser($userId)
+    {
+        try {
+            $filters = [
+                [
+                    'column' => 'status',
+                    'columnSqlTranslate' => ':status',
+                    'operator' => '!=',
+                    'value' => TaskLogInterface::STATUS_ARCHIVED
+                ],
+                [
+                    'column' => 'owner',
+                    'columnSqlTranslate' => ':owner',
+                    'operator' => '=',
+                    'value' => $userId
+                ],
+            ];
+
+            $qb = $this->getQueryBuilder()
+                ->select('*')
+                ->from($this->getTableName());
+
+            foreach ($filters as $filter) {
+                $qb->andWhere($filter['column'] . $filter['operator'] . $filter['columnSqlTranslate'])
+                    ->setParameter($filter['column'], $filter['value'])
+                ;
+            }
+
+            $rows = $qb->execute()->fetchAll();
+            $collection = TaskLogCollection::createFromArray($rows);
```

Comment:
So, as I said, I would remove TaskLogCollection and simple returning the result as an array. Collection is a nice thing in Doctrine ORM but unfortunately we don't use ORM but only DBAL.

Reply:
It's not a matter of doctrine ORM , it's just a collection :).